pkg/services/ngalert/schedule/schedule.go

```diff
@@ -162,11 +171,13 @@ func (sch *schedule) DeleteAlertRule(keys ...ngmodels.AlertRuleKey) {
 		// Delete the rule routine
 		ruleInfo, ok := sch.registry.del(key)
 		if !ok {
-			sch.log.Info("Alert rule cannot be stopped as it is not running", key.LogContext()...)
-			continue
+			if err != nil {
+				sch.log.Info("Alert rule cannot be stopped as it is not running", key.LogContext()...)
```

Comment:
Wouldn't we want to log this error message for paused rules too?

Reply:
I enabled it at first but it was outputting this message every evaluation interval. It was kind of annoying

Reply:
Also, It wouldn't make sense for me to have a recurring message telling me that the alert I paused can't be paused because it is paused.

Reply:
Why would it be recurring? It should only happen once when you attempt to pause a rule that isn't running?

Reply:
It is recurring because the DB query retrieves paused and unpaused alerts and just tries to pause the ones that have `IsPaused` field to `true`. You can't know where it was already paused or not.

Reply:
Hmm, this doesn't sound right to me? We should be able to know if an alert rule is running? I don't think we should settle for just pausing all paused alert rules on every iteration of the scheduler, not even because of efficiency, but it suggests to me that are are working around the internal design of our scheduler.

Reply:
Hmm, this doesn't sound right to me? We should be able to know if an alert rule is running? I don't think we should settle for just pausing all paused alert rules on every iteration of the scheduler, not because of efficiency, but because it suggests to me that are are working around the internal design of our scheduler.